### PR TITLE
feat(core): return cesium credit conditionally in getCredits

### PR DESCRIPTION
**Pull request by ZTongci**
## Overview

The Cesium-ion logo was always shown in the credit display, even when the scene contained no Cesium-ion assets. `creditDisplay._currentCesiumCredit` is populated by Cesium as soon as the viewer initialises, regardless of whether any Ion asset is actually loaded.

This change computes whether Ion assets are in use **inside the core's `Map` component** — the only place that already has access to both `property` (ViewerProperty) and `layers` (Layer[]) — and threads the result down to `getCredits()` so the credit is only emitted when Ion assets are genuinely present.

## What I've done

- **`src/Map/cesiumIonDetection.ts`** (new) — `computeHasCesiumIonAsset(property, layers)` detects Ion usage across:
  - **Tiles**: `type` starts with `cesium_ion`, or is a legacy alias (`default`, `default_road`, `default_label`, `black_marble`).
  - **Terrain**: type `cesium` or `cesiumion` with `enabled: true`, or `assets.cesium.terrain.ionUrl` contains `ion.cesium.com`.
  - **Layers**: `osm-buildings`, `google-photorealistic` not routed through Google Maps API, or `3dtiles` with an `ion.cesium.com` URL. Recurses into `LayerGroup.children`.
- **`src/Map/index.tsx`** — Computes `hasCesiumIonAsset` internally via `useMemo` and passes it to the engine. `hasCesiumIonAsset` is excluded from `Map.Props` via `Omit` so external callers cannot set it.
- **`src/Visualizer/index.tsx`** — Removed `hasCesiumIonAsset` from `CoreVisualizerProps`; callers no longer need to know about Ion detection.
- **`src/engines/Cesium/common.ts`** — `getCredits(viewer, hasCesiumIonAsset?)` sets `engine.cesium` to `undefined` when `hasCesiumIonAsset === false`. `undefined` (not provided) preserves existing behavior.
- **`src/engines/Cesium/hooks.ts`** — Accepts `hasCesiumIonAsset`, stores it in a ref, passes ref to `useEngineRef` and value to `getCredits`.
- **`src/engines/Cesium/index.tsx`** — Destructures and forwards `hasCesiumIonAsset` to `useHooks`.
- **`src/engines/Cesium/hooks/useEngineRef.ts`** — Passes `hasCesiumIonAssetRef.current` to `getCredits`.

## How I tested

Manually verified: default scene (OSM tile + reearth terrain) → logo hidden; switching to a Cesium Ion tile → logo appears; switching back → logo disappears after the next 3-second credit poll.

## Backward compatibility

`hasCesiumIonAsset === false` is the only case that suppresses the credit. When the flag computes to `undefined` or `true`, the existing behavior is preserved.

### CODE REVIEW

File: src/Map/cesiumIonDetection.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+function layerUsesIon(layer: LayerSimple): boolean {
+  const data = layer.data;
+  if (!data) return false;
+  if (data.type === "osm-buildings") return true;
+  if (data.type === "google-photorealistic") {
+    if (data.provider === "reearth") return false;
+    if (data.serviceTokens?.googleMapApiKey && data.provider !== "cesium-ion") return false;
+    return true;
+  }
+  if (data.type === "3dtiles" && isIonUrl(data.url)) return true;
```
Comment (airslice) [inline src/Map/cesiumIonDetection.ts]: Maybe we could remove the type limit here.

File: src/Map/cesiumIonDetection.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  return false;
+}
+
+function layerUsesIon(layer: LayerSimple): boolean {
+  const data = layer.data;
+  if (!data) return false;
+  if (data.type === "osm-buildings") return true;
+  if (data.type === "google-photorealistic") {
+    if (data.provider === "reearth") return false;
+    if (data.serviceTokens?.googleMapApiKey && data.provider !== "cesium-ion") return false;
```
Comment (airslice) [inline src/Map/cesiumIonDetection.ts]: I think we only need to check the provider here. 
On the implement of `googleMapPhotorealisticResource`, maybe we don't need the fallback to cesium-ion when there's no googleMapApiKey.

File: src/Map/cesiumIonDetection.ts (hunk 1 above)
Comment (ZTongci) [reply src/Map/cesiumIonDetection.ts]: Fixed! Please check.

File: src/Map/cesiumIonDetection.ts (hunk 2 above)
Comment (ZTongci) [reply src/Map/cesiumIonDetection.ts]: Fixed! Please check.

File: src/Map/cesiumIonDetection.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  return false;
+}
+
+function layerUsesIon(layer: LayerSimple): boolean {
+  const data = layer.data;
+  if (!data) return false;
+  if (data.type === "osm-buildings") return true;
+  if (data.type === "google-photorealistic") {
+    return data.provider === "cesium-ion";
+  }
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply src/Map/cesiumIonDetection.ts:38]: No change needed here. The detection intentionally only checks `provider === "cesium-ion"` as suggested by the human reviewer. The `!googleMapApiKey` fallback to Ion in the implementation is considered legacy behavior that should be removed separately. The detection reflects the intended direction.

File: src/Map/cesiumIonDetection.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    return layer.children.some(anyLayerUsesIon);
+  }
+  return layerUsesIon(layer);
+}
+
+export function computeHasCesiumIonAsset(property?: ViewerProperty, layers?: Layer[]): boolean {
+  if (property?.tiles?.some(tileUsesIon)) return true;
+  if (terrainUsesIon(property)) return true;
+  if (layers?.some(anyLayerUsesIon)) return true;
+  return false;
```
[the remark this answers is not in the document]
Comment (ZTongci) [reply src/Map/cesiumIonDetection.ts:54]: Good point! Added a unit test file `src/Map/cesiumIonDetection.test.ts` with 25 tests covering tiles, terrain, layers (osm-buildings, google-photorealistic, ion URL), LayerGroup recursion, and edge cases. All tests pass.

**Review by bnimit — DISMISSED** [review]
LGTM!

**Review by airslice — APPROVED** [review]
LGTM!
